src/ResourceManager/PowerBI/Commands.PowerBI/Commands/RemoveAzureRmPowerBIEmbeddedCapacity.cs

```diff
@@ -0,0 +1,111 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Management.Automation;
+using System.Security.Permissions;
+using Microsoft.Azure.Commands.PowerBI.Models;
+using Microsoft.Azure.Commands.PowerBI.Properties;
+using Microsoft.Azure.Commands.PowerBI.Utilities;
+using Microsoft.Azure.Management.PowerBIDedicated.Models;
+using Microsoft.Rest.Azure;
+
+namespace Microsoft.Azure.Commands.PowerBI
+{
+    [Cmdlet(VerbsCommon.Remove, "AzureRmPowerBIEmbeddedCapacity", SupportsShouldProcess = true), 
+        OutputType(typeof(PSDedicatedCapacity))]
+    public class RemovePowerBIEmbeddedCapacity : PowerBICmdletBase
+    {
+        protected const string CmdletParametersSet = "ByNameAndResourceGroup";
+        protected const string ObjectParameterSet = "ByInputObject";
+        protected const string ResourceIdParameterSet = "ByResourceId";
+
+        [Parameter(
+            ParameterSetName = CmdletParametersSet,
+            Position = 0, 
+            Mandatory = true,
+            HelpMessage = "Name of capacity to be removed.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            ParameterSetName = CmdletParametersSet,
+            Mandatory = false,
+            HelpMessage = "Name of resource group under which the capacity exists.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            ParameterSetName = ResourceIdParameterSet,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipelineByPropertyName = true, 
+            HelpMessage = "PowerBI Embedded Capacity ResourceID.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            ParameterSetName = ObjectParameterSet,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            HelpMessage = "PowerBI Embedded Capacity object.")]
+        [ValidateNotNullOrEmpty]
+        public PSDedicatedCapacity InputObject { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        [PermissionSet(SecurityAction.Demand, Name = "FullTrust")]
+        public override void ExecuteCmdlet()
+        {
+            string capacityName = string.Empty;
```

Comment:
@selasagi couldn't you just assign this to `Name` and avoid having the if-statement below where you check if `Name` is null or empty? If a user provides `ResourceId` or `InputObject`, these values will get overridden anyway.

Reply:
fixed